static/app/views/starfish/modules/databaseModule/queries.ts

```diff
@@ -181,40 +181,83 @@ export const useQueryTopDbOperationsChart = (
   });
 };
 
-export const useGetTransactionsForTables = (tableNames: string[], interval: number) => {
+type TopTransactionData = {
+  interval: string;
+  transaction: string;
+  epm?: number;
+  p75?: number;
+};
+
+export const useGetTransactionsForTables = (
+  tableNames: string[],
+  interval: number
+): DefinedUseQueryResult<TopTransactionData[]> => {
   const pageFilter = usePageFilters();
   const {startTime, endTime} = getDateFilters(pageFilter);
   const dateFilters = getDateQueryFilter(startTime, endTime);
-  const transactionFilter = `transaction IN (${getTransactionsFromTableSubquery(
-    tableNames,
-    dateFilters
-  )})`;
 
-  const filters = [transactionFilter];
-
-  const query = `
-    SELECT
-      transaction,
-      floor(quantile(0.75)(exclusive_time), 5) as p75,
-      count() as count,
-      toStartOfInterval(start_timestamp, INTERVAL ${interval} hour) as interval
-    FROM default.spans_experimental_starfish
-    WHERE
-      ${filters.join(' AND ')}
-      ${dateFilters}
-    GROUP BY interval, transaction
-    ORDER BY interval
-  `;
+  const transactionNameQuery = getTransactionsFromTableSubquery(tableNames, dateFilters);
 
-  const {start, end} = pageFilter.selection.datetime;
+  const {start, end, period} = pageFilter.selection.datetime;
 
-  return useQuery({
+  const result1 = useQuery<{transaction: string}[]>({
     enabled: !!tableNames?.length,
-    queryKey: ['topTable', tableNames.join(','), start, end],
-    queryFn: () => fetch(`${HOST}/?query=${query}`).then(res => res.json()),
+    queryKey: ['topTransactionNames', tableNames.join(','), start, end],
+    queryFn: () =>
+      fetch(`${HOST}/?query=${transactionNameQuery}`).then(res => res.json()),
     retry: false,
     initialData: [],
   });
+
+  const queryParams = [
+    'dataset=metricsEnhanced',
+    'excludeOther=1',
+    'field=transaction',
+    `field=epm%28%29`,
+    `field=p75%28transaction.duration%29`,
+    `interval=${interval}h`,
+    'partial=1',
+    `query=${encodeURIComponent(
+      `transaction:[${result1.data?.map(d => d.transaction).join(',')}]`
+    )}`,
+    `statsPeriod=${period}`,
+    'topEvents=5',
+    `yAxis=epm%28%29`,
+    `yAxis=p75%28transaction.duration%29`,
+    'project=1',
+  ].join('&');
```

Comment:
Can we use the EventView DiscoverQuery stuff here? Part of the reason i'm saying we should use metrics is cause that's the actual dataset we'll use in the final version

Reply:
Yeah, that's a good idea, I was more or less copying the other example of metrics usage in the starfish project, but i'll make that change too cause it's final.